app/(product)/public-cloud/(product)/aws-roles/[licencePlate]/[role]/page.tsx

```diff
@@ -6,6 +6,47 @@ import TableBodyAWSRoles from '@/components/table/TableBodyAWSRoles';
 import { capitalizeFirstLetter } from '@/components/utils/capitalizeFirstLetter';
 import { useQuery } from '@tanstack/react-query';
 import { useSearchParams, useParams } from 'next/navigation';
+import UserAWSRolesTableTop from '@/components/table/TableTopUserAWSRoles';
+import AddUserModal from '@/components/modal/AddUser';
+import { useEffect, useState } from 'react';
+import DeleteUserModal from '@/components/modal/DeleteUser';
+import EmptyBody from '@/components/EmptyUsersList';
+
+async function addUser(userEmail: string, groupId: string): Promise<string | undefined> {
+  const url = `/api/public-cloud/aws-roles/addUser?userEmail=${userEmail}&groupId=${groupId}`;
+  try {
+    const response = await fetch(url, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    if (response.ok) {
+      return response.statusText;
+    }
+    console.error('Failed to handle PUT request:', response.statusText);
+  } catch (error) {
+    console.error('Error during PUT request:', error);
+  }
+}
+
+async function deleteUser(userId: string, groupId: string): Promise<string | undefined> {
```

Comment:
Can we move all query services to `services` directory using `Axios` to separate `service` layer from the main page logic?
https://github.com/bcgov/platform-services-registry/blob/main/services/security-config.ts

and let's have `@tanstack/react-query` to handle errors.

Reply:
all aws roles queries moved to services `@tanstack/react-query` used to handle errors